doc/manual/manual.tex

```diff
@@ -1595,14 +1741,198 @@ \subsubsection{Compressible models}
+\item Formulating the buoyancy term in the material model to be used on the right-hand side of the momentum equation, 
+\item Prescribing a suitable reference state for the temperature, pressure, and density, the adiabatic profile.
+\end{enumerate}
+
+\subsubsection{Mass conservation approximation}
+
+First, we have to choose how to approximate the conservation of mass: $\nabla \cdot (\rho \mathbf u) = 0$ (see Equation~\eqref{eq:stokes-2}). 
+We provide the following options:
+
+\begin{itemize}
+
+\item
+``incompressible'':
+\[
+ \nabla \cdot \textbf{u} = 0,
+\]
+
+\item
+``isothermal compression'':
+\[
+ \nabla \cdot \textbf{u} = -\frac{1}{\rho} \frac{\partial \rho}{\partial p} \rho \textbf{g} \cdot \textbf{u},
+\]
+where the compressibility $\kappa = \frac{1}{\rho} \frac{\partial \rho}{\partial p}$ 
+is defined in the material model. 
+
+\item
+``reference density profile'':
+\[
+ \nabla \cdot \textbf{u} = -\frac{1}{\bar{\rho}} \frac{\partial \bar{\rho}}{\partial z} \frac{\textbf{g}}{\|\textbf{g}\|} \cdot \textbf{u},
+\]
+where reference profiles for the density $\bar{\rho}$ and the density gradient $\frac{\partial \bar{\rho}}{\partial z}$
+provided by \text{AdiabaticConditions}
+are used. Note that the
+gravity is assumed to point downwards in depth direction
+
+\item
+``implicit reference density profile'':
+\[
+ \nabla \cdot \textbf{u} + \frac{1}{\bar{\rho}} \frac{\partial \bar{\rho}}{\partial z} \frac{\textbf{g}}{g} \cdot \textbf{u} = 0,
+\]
+which uses the same approximation for the density as ``reference density profile'', 
+but implements this term on the left-hand side instead of the right-hand side of the mass
+conservation equation. This effectively uses the current velocity $\textbf{u}$ instead
+of an extrapolated velocity from the last timesteps.
+
+ \item 
+``ask material model'', which uses ``isothermal compression'' if the material model is compressible 
+(as reported by the \texttt{is\_compressible()} function) and ``incompressible'' otherwise.
+\end{itemize}
+
+
+\paragraph*{On the stress tensor approximation.}
+
+Incompressibility in the mass conservation equation automatically simplifies the shear strain rate in the momentum and temperature equation from
+\[
+ \nabla \cdot \tau = 
+ \nabla \cdot \left[
+ 2\eta \left(\varepsilon(\mathbf u)
+                                  - \frac{1}{3}(\nabla \cdot \mathbf u)\mathbf 1\right)
+                                  \right]
+\]
+to
+\[
+  \nabla \cdot \tau =  \nabla \cdot \left[
+ 2\eta \varepsilon(\mathbf u) \right].
+\]
+
+\subsubsection{Temperature equation approximation}
+
+The density occurs multiple times in the temperature equation. Depending on the selected approximation it is computed in two different ways:
+
+\begin{itemize}
+ \item 
+``real density'': Use the full density as computed for the buoyancy term of the momentum equation (i.e. the return value of the material model for ``density``),
```

Comment:
Maybe state this as "Use the full density $\rho(p,T)$ that equals the one also used in the buoyancy term of the force balance equation; this is also the value that is computed by the material models when asked for the density."